src/dodal/plans/scanspec.py

```diff
@@ -0,0 +1,79 @@
+import operator
+from functools import reduce
+from typing import Any, List, Mapping, Optional
+
+import bluesky.plans as bp
+from bluesky.protocols import Movable, Readable
+from cycler import Cycler, cycler
+from scanspec.specs import Spec
+
+from dodal.common import MsgGenerator
+
+"""
+Plans related to the use of the `ScanSpec https://github.com/dls-controls/scanspec`
+library for constructing arbitrarily complex N-dimensional trajectories, similar to
+Diamond's "mapping scans" using ScanPointGenerator.
+"""
+
+
+def scan(
+    detectors: List[Readable],
+    axes_to_move: Mapping[str, Movable],
+    spec: Spec[str],
+    metadata: Optional[Mapping[str, Any]] = None,
+) -> MsgGenerator:
+    """
+    Scan wrapping `bp.scan_nd`
+
+    Args:
+        detectors: List of readable devices, will take a reading at
+                                    each point
+        axes_to_move: All axes involved in this scan, names and
+            objects
+        spec: ScanSpec modelling the path of the scan
+        metadata: Key-value metadata to include
+                                                          in exported data, defaults to
+                                                          None.
```

Comment:
Nit: Not sure what's happening with whitespace here?

Reply:
Fixed